plugins/dbviewer/api/api.js

```diff
@@ -120,14 +120,20 @@ var plugin = {},
                     });
                 }
                 
-                async.map(apps, getEvents, function (err, events) {
+                // if app_id provided, update app list
+                if (params.qstring.app_id) {
+                    var filteredApps = apps.filter((app) => app._id+"" === params.qstring.app_id+"");    
+                } else {
+                    var filteredApps = apps;
+                }
+                
+                async.map(filteredApps, getEvents, function (err, events) {
```

Comment:
I don't know if we need a filter here, I think everywhere where we call this method, **apps** array has been already prefiltered accordingly, at least it should be. But if it helps you sleep at night, sure :)